diff --git a/node_modules/@vitejs/plugin-react/dist/index.mjs b/node_modules/@vitejs/plugin-react/dist/index.mjs
index e61f42b..a6ad925 100644
--- a/node_modules/@vitejs/plugin-react/dist/index.mjs
+++ b/node_modules/@vitejs/plugin-react/dist/index.mjs
@@ -202,6 +202,19 @@ function viteReact(opts = {}) {
       });
       if (result) {
         let code2 = result.code;
+        if (filepath.includes('graphiql-react')) {
+          if (
+            /^import \{ c as _c } from "/m.test(code2)
+          ) {
+            console.info('🚀 File', filepath, 'was optimized with react-compiler')
+          } else if (!/^'use no memo'/m.test(code2)) {
+            console.error('❌ File', filepath, 'was not optimized with react-compiler')
+            console.log(code2)
+            if (process.env.NODE_ENV === 'production') {
+              process.exit(1)
+            }
+          }
+        }
         if (useFastRefresh) {
           if (refreshContentRE.test(code2)) {
             code2 = addRefreshWrapper(code2, id);
