diff --git a/node_modules/graphql/execution/execute.js b/node_modules/graphql/execution/execute.js
index eb56e4b..5a4a398 100644
--- a/node_modules/graphql/execution/execute.js
+++ b/node_modules/graphql/execution/execute.js
@@ -2055,7 +2055,7 @@ async function getNextAsyncStreamItemResult(
     getNextAsyncStreamItemResult(
       streamItemQueue,
       streamPath,
-      index,
+      index + 1,
       asyncIterator,
       exeContext,
       fieldGroup,
diff --git a/node_modules/graphql/execution/execute.mjs b/node_modules/graphql/execution/execute.mjs
index 641c067..c7641d4 100644
--- a/node_modules/graphql/execution/execute.mjs
+++ b/node_modules/graphql/execution/execute.mjs
@@ -1984,7 +1984,7 @@ async function getNextAsyncStreamItemResult(
     getNextAsyncStreamItemResult(
       streamItemQueue,
       streamPath,
-      index,
+      index + 1,
       asyncIterator,
       exeContext,
       fieldGroup,
