diff --git a/node_modules/js-green-licenses/build/src/checker.js b/node_modules/js-green-licenses/build/src/checker.js
index 79a88a3..2f8a355 100644
--- a/node_modules/js-green-licenses/build/src/checker.js
+++ b/node_modules/js-green-licenses/build/src/checker.js
@@ -229,7 +229,7 @@ class LicenseChecker extends events_1.EventEmitter {
         if (this.failedPackages.has(spec))
             return;
         // remove tilde/caret to check for an exact version, ^0.5.0-rc.0 becomes 0.5.0-rc.0
-        const version = versionSpec.replace(/^[^~]/, '');
+        const version = versionSpec.replace(/^[~^]/, '');
         // if the dependency is a local package then skip verification at this step. will be checked independently
         if (this.localPackages.has(`${packageName}@${version}`))
             return;
